### DISCUSSION TITLE
Default PML round trip Absorption for single-precision builds

### DISCUSSION THREAD

**Discussion by smartalecH** [NanoComp/meep#2279] [Q&A] [answered]
Currently, the default [`R_asymptotic` for the PML region](https://meep.readthedocs.io/en/latest/Python_User_Interface/#pml) is $10^{-15}$. This was probably chosen assuming meep was built with double precision. Should we expect any issues with this default value when building meep with single precision?

**Comment by oskooi** [comment, marked as answer]
Note that there are various C++ unit tests in [tests/pml.cpp](https://github.com/NanoComp/meep/blob/master/tests/pml.cpp) which verify the PML reflection scaling with length and resolution for isotropic and anisotropic materials in Cartesian and cylindrical coordinates. These tests also involve a hard-coded value of 1e-15 for the round-trip reflection:

https://github.com/NanoComp/meep/blob/9abe44e0818780223d712d891c1744ef19073916/tests/pml.cpp#L10

Obviously, these tests pass the CI via Github Actions which includes building with single precision:

https://github.com/NanoComp/meep/blob/9abe44e0818780223d712d891c1744ef19073916/.github/workflows/build-ci.yml#L154-L160

More generally, to obtain PML reflections (due to the discretization) that are smaller than ~1e-8 (i.e., machine epsilon for single-precision floating point) one would typically need to go to very fine grid resolutions.

**Reply by smartalecH** [reply]
Great! Thanks, @oskooi.
